patches/kex_rlwe_newhope_avx2/patch1.patch

```diff
@@ -0,0 +1,1073 @@
++extern void OQS_chacha_avx2(const unsigned char *k, const unsigned char *n, const unsigned char *in, unsigned char *out, size_t inlen, size_t rounds);
+ 
+-int crypto_stream_chacha20(unsigned char *out, unsigned long long outlen, const unsigned char *n, const unsigned char *k) 
++int OQS_crypto_stream_chacha20(unsigned char *out, unsigned long long outlen, const unsigned char *n, const unsigned char *k)
+ {
+-	chacha_avx2(k, n, NULL, out, (size_t)outlen, 20);
++	OQS_chacha_avx2(k, n, NULL, out, (size_t)outlen, 20);
+ 	return 0;
+ }
+diff --git a/src/kex_rlwe_newhope/avx2/error_correction.c b/src/kex_rlwe_newhope/avx2/error_correction.c
+index 0970e6f..eaf2952 100644
+--- a/src/kex_rlwe_newhope/avx2/error_correction.c
++++ b/src/kex_rlwe_newhope/avx2/error_correction.c
+@@ -3,7 +3,7 @@
+ 
+ //See paper for details on the error reconciliation
+ 
+-extern void hr(poly *c, const poly *v, unsigned char rand[32]);
++extern void OQS_hr(poly *c, const poly *v, unsigned char rand[32]);
+ 
+ void helprec(poly *c, const poly *v, const unsigned char *seed, unsigned char nonce)
+ {
+@@ -17,5 +17,5 @@ void helprec(poly *c, const poly *v, const unsigned char *seed, unsigned char no
+ 
+   crypto_stream(rand,32,n,seed);
+ 
+-  hr(c, v, rand);
++  OQS_hr(c, v, rand);
+ }
+diff --git a/src/kex_rlwe_newhope/avx2/error_correction.h b/src/kex_rlwe_newhope/avx2/error_correction.h
+index 488c105..c0e913d 100644
+--- a/src/kex_rlwe_newhope/avx2/error_correction.h
++++ b/src/kex_rlwe_newhope/avx2/error_correction.h
+@@ -9,7 +9,7 @@
+ #include "poly.h"
+ #include <stdio.h>
+ 
+-void helprec(poly *c, const poly *v, const unsigned char *seed, unsigned char nonce);
+-void rec(unsigned char *key, const poly *v, const poly *c);
++static void helprec(poly *c, const poly *v, const unsigned char *seed, unsigned char nonce);
++void OQS_rec(unsigned char *key, const poly *v, const poly *c);
+ 
+ #endif
+diff --git a/src/kex_rlwe_newhope/avx2/fips202.c b/src/kex_rlwe_newhope/avx2/fips202.c
+index f649a7d..b83029c 100644
+--- a/src/kex_rlwe_newhope/avx2/fips202.c
++++ b/src/kex_rlwe_newhope/avx2/fips202.c
+@@ -60,7 +60,7 @@ static const uint64_t KeccakF_RoundConstants[NROUNDS] =
+     (uint64_t)0x8000000080008008ULL
+ };
+ 
+-void KeccakF1600_StatePermute(uint64_t * state)
++static void KeccakF1600_StatePermute(uint64_t * state)
+ {
+   int round;
+ 
+@@ -381,19 +381,19 @@ static void keccak_squeezeblocks(unsigned char *h, unsigned long long int nblock
+ }
+ 
+ 
+-void shake128_absorb(uint64_t *s, const unsigned char *input, unsigned int inputByteLen)
++static void shake128_absorb(uint64_t *s, const unsigned char *input, unsigned int inputByteLen)
+ {
+   keccak_absorb(s, SHAKE128_RATE, input, inputByteLen, 0x1F);
+ }
+ 
+ 
+-void shake128_squeezeblocks(unsigned char *output, unsigned long long nblocks, uint64_t *s)
++static void shake128_squeezeblocks(unsigned char *output, unsigned long long nblocks, uint64_t *s)
```

Comment:
I'm confused, some functions are `static` while others are renamed with `OQS_`. Is there a reason for this?

Reply:
All the function which are public and part of API and of course implemented in assembly have to be OQS_ prefixed.